Emails: Show labels in add mailbox in Google Workspace

AllTerrainDeveloper opened the pull request:
#### Changes proposed in this Pull Request

Show labels in the form to add mailboxes for Google Workspaces

#### Testing instructions

**Adding new mailbox**
BEFORE | AFTER
------------ | -------------
![image](https://user-images.githubusercontent.com/5689927/128860180-862ea028-a388-45c2-bd68-e5bf746aad2a.png) | ![image](https://user-images.githubusercontent.com/5689927/128860314-04dce6cd-6734-4beb-94ee-af705d990b4d.png)

**Adding additional mailboxes**
BEFORE | AFTER
------------ | -------------
<img alt="Screenshot 2021-08-11 at 14 20 26" width="1060" src="https://user-images.githubusercontent.com/3376401/129028779-62c05f3e-5904-4537-b781-8ac96a4e958a.png"> | ![image](https://user-images.githubusercontent.com/5689927/129167210-4ee807fe-c08a-4607-9dd6-49555f15a3d3.png)


Related to {1200182182542585-as-1200722335695079}

AllTerrainDeveloper committed:
Emails: Show labels in add mailbox in Google Workspace

AllTerrainDeveloper committed:
Emails: Lint fixes

daledupreez commented on client/components/gsuite/docs/new-user-list.tsx:
```diff
@@ -59,6 +59,7 @@ const GSuiteNewUserListExample = (): React.FunctionComponent => {
 				onUsersChange={ ( changedUsers ) => setUsers( changedUsers ) }
 				users={ users }
 				onReturnKeyPress={ () => void 0 }
+				showLabels={ true }
```
Can we remove the flag rather than switching the value of the prop?

Our long-term goal has been to show labels in all cases, so making that be the only option is going to be simpler for us in the long run.

daledupreez commented on client/my-sites/email/google-workspace-add-users/index.jsx:
```diff
@@ -52,7 +52,7 @@ import { getProductsList } from 'calypso/state/products-list/selectors/get-produ
 import './style.scss';
 
-class GSuiteAddUsers extends React.Component {
+class GoogleWorkspaceAddUsers extends React.Component {
```
These components are still in use for existing G Suite customers, so I am not sure that a refactor/rename makes sense yet. It may also be better to push that into a separate PR.

daledupreez commented on client/my-sites/email/controller.js:
```diff
@@ -10,7 +10,7 @@ import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopp
 import EmailManagementHome from 'calypso/my-sites/email/email-management/email-home';
 import EmailProvidersComparison from 'calypso/my-sites/email/email-providers-comparison';
-import GSuiteAddUsers from 'calypso/my-sites/email/gsuite-add-users';
+import GSuiteAddUsers from 'calypso/my-sites/email/google-workspace-add-users';
```
This import should probably be updated as well.

daledupreez reviewed (changes requested):
I think we should remove the `showLabels` flag entirely.

It may also be cleaner to handle the rename in a separate PR. My initial concern on that front (without looking too closely) is that some components may have been targeting the old class names via CSS.

stephanethomas commented:
> I have also renamed the containing folder for the component, to be related with Google Workspaces instead of gsuite related

Just so you know @AllTerrainDeveloper, we've toyed with the idea of renaming occurrences of G Suite with Google Workspace when we were working on the migration to Google Workspace. We ultimately decided that we would only update mentions that are visible to users, and would keep referring to G Suite in our code (except in a few places where we needed a specific behavior for Google Workspace): the amount of work required for that change would indeed be far too important, especially in regard of its very limited benefits. You are welcome to rename any occurrence of the former name of G Suite though, which is Google Apps 🙂.

AllTerrainDeveloper committed:
Emails: PR suggestions (reverse renaming, remove flag)

AllTerrainDeveloper committed:
Emails: Fix Linting

AllTerrainDeveloper commented:
I've just reverted the renaming and removed the flags as discussed 🙂

AllTerrainDeveloper commented:
The test that fails is "Activate and Preview theme" seems unrelated and it typically fails with other builds

daledupreez commented on client/components/gsuite/gsuite-new-user-list/new-user.tsx:
```diff
@@ -128,12 +121,10 @@ const GSuiteNewUser: FunctionComponent< Props > = ( {
-		<div
-			className={ classNames( 'gsuite-new-user-list__new-user', { 'show-labels': showLabels } ) }
-		>
+		<div className={ classNames( 'gsuite-new-user-list__new-user' ) }>
```
We can't remove the `show-labels` class without updating the style that is dependent on the class:
https://github.com/Automattic/wp-calypso/blob/update/labels-in-add-new-mailboxes-for-google-workspace/client/components/gsuite/gsuite-new-user-list/style.scss#L66

daledupreez reviewed (changes requested):
Could you remove the following from the PR description? You removed these changes from the PR. 😁 
> As part of this change, I have also renamed the containing folder for the component, to be related with Google Workspaces instead of gsuite related

Could you also clean up/remove the `show-label` CSS rule? I don't think it's needed any more.

I think it would also be good to be more explicit about what you think falls inside the scope of testing vs not. I found the following behaviour when adding multiple Google mailboxes:
<img width="1060" alt="Screenshot 2021-08-11 at 14 20 26" src="https://user-images.githubusercontent.com/3376401/129028779-62c05f3e-5904-4537-b781-8ac96a4e958a.png">

I _think_ this was broken by the vertical display work a few weeks ago, but is also impacted by the missing class I noticed. (Adding the class back just breaks things differently! 🙃) Could you fix that in a follow-up PR?

AllTerrainDeveloper commented:
@daledupreez I guess the "Remove user" button was originally at the end of the form, as per the new screenshot, right?

AllTerrainDeveloper committed:
Emails: Pull Request suggestions
Remove class from CSS file
Move the "Remove user" button to the bottom of the form

daledupreez reviewed (approved):
LGTM! 🚢
